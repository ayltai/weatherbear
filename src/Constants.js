export const Constants = {};

Constants.TYPE_DARK_SKY     = 0;
Constants.TYPE_ACCU_WEATHER = 1;
Constants.TYPE_WEATHER_BIT  = 2;

Constants.UNIT_SI       = 'si';
Constants.UNIT_IMPERIAL = 'imperial';
