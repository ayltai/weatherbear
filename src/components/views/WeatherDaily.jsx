import { Grid, Tooltip, } from '@material-ui/core';
import moment from 'moment';
import PropTypes from 'prop-types';
import React, { Fragment, } from 'react';

import { IconHelper, } from '../../utils/IconHelper';
import { MeasurementHelper, } from '../../utils/MeasurementHelper';
import { NumberFormatHelper, } from '../../utils/NumberFormatHelper';
import { Constants, } from '../../Constants';
import { ShadowedText, } from './ShadowedText';
import { Temperature, } from './Temperature';

export const WeatherDaily = props => {
    let i = 0;

    const date = (
        <ShadowedText
            key={i++}
            variant='body2'
            align='center'
            noShadow={props.noShadow}
            noWrap>
            {moment(props.date || new Date()).format('ddd')}
        </ShadowedText>
    );

    const icon = (
        <ShadowedText
            key={i++}
            variant='h4'
            align='center'
            noShadow={props.noShadow}
            noWrap>
            <i className={`wi ${IconHelper.toIcon(props.iconId, props.iconType || Constants.TYPE_DARK_SKY)}`}></i>
        </ShadowedText>
    );

    const temperatureHigh = (
        <Temperature
            key={i++}
            variant='body2'
            align='center'
            prefix={`${props.temperatureHighLabel || '🔥'} `}
            unit={props.unit}
            noShadow={props.noShadow}
            noWrap>
            {props.temperatureHigh}
        </Temperature>
    );

    const temperatureLow = (
        <Temperature
            key={i++}
            variant='body2'
            align='center'
            prefix={`${props.temperatureLowLabel || '❄'} `}
            unit={props.unit}
            noShadow={props.noShadow}
            noWrap>
            {props.temperatureLow}
        </Temperature>
    );

    const precip = (
        <ShadowedText
            key={i++}
            variant='caption'
            align='center'
            noShadow={props.noShadow}
            noWrap>
            {`${props.precipLabel || '🌧'}${Math.round(props.precipProbability * 100)}%`}
        </ShadowedText>
    );

    const blocks = [];

    if (props.orientation === 'vertical') {
        blocks.push(date);
        blocks.push(icon);
        blocks.push(temperatureHigh);
        blocks.push(temperatureLow);
        blocks.push(precip);
    } else {
        blocks.push(
            <Grid
                key={i++}
                item
                xs={2}>
                {date}
                {precip}
            </Grid>
        );

        blocks.push(
            <Grid
                key={i++}
                item
                xs={2}>
                {icon}
            </Grid>
        );

        blocks.push(
            <Grid
                key={i++}
                item
                xs={6}>
                <ShadowedText
                    variant='caption'
                    align='left'
                    noShadow={props.noShadow}
                    noWrap>
                    {props.summary}
                </ShadowedText>
            </Grid>
        );

        blocks.push(
            <Grid
                key={i}
                item
                xs={2}>
                {temperatureHigh}
                {temperatureLow}
            </Grid>
        );
    }

    return (
        <Tooltip title={
            <Fragment>
                {moment(props.date).format('LL')}<br />
                {props.summary}<br />
                {`${props.temperatureLabel || '🌡'} ${MeasurementHelper.toTemperature(props.temperatureLow, props.unit)} - ${MeasurementHelper.toTemperature(props.temperatureHigh, props.unit)}`}<br />
                {`${props.precipLabel || '🌧'} ${NumberFormatHelper.toFixed(100 * props.precipProbability, 0)}% ${NumberFormatHelper.toFixed(props.precipIntensity)}mm`}<br />
                {props.humidity ? (
                    <>
                        {`${props.humidityLabel || '🌢'} ${NumberFormatHelper.toFixed(100 * props.humidity, 0)}`}<br />
                    </>
                ) : null}
                {`${props.windSpeedLabel || '💨'} ${NumberFormatHelper.toFixed(MeasurementHelper.toSpeed(props.windSpeed, props.unit), 0)}${props.windSpeedSuffix}`}<br />
                {`${props.uvIndexLabel || '☀'} ${props.uvIndex}`}
            </Fragment>
        }>
            <Grid
                container
                direction={props.orientation === 'vertical' ? 'column' : 'row'}
                alignItems='center'
                justify='space-evenly'>
                {blocks}
            </Grid>
        </Tooltip>
    );
};

WeatherDaily.propTypes = {
    orientation          : PropTypes.oneOf([
        'horizontal',
        'vertical',
    ]),
    date                 : PropTypes.object,
    summary              : PropTypes.string,
    iconId               : PropTypes.oneOfType([
        PropTypes.number,
        PropTypes.string,
    ]),
    iconType             : PropTypes.oneOf([
        Constants.TYPE_ACCU_WEATHER,
        Constants.TYPE_DARK_SKY,
        Constants.TYPE_WEATHER_BIT,
    ]),
    temperatureLabel     : PropTypes.string,
    temperatureHigh      : PropTypes.number,
    temperatureHighLabel : PropTypes.string,
    temperatureLow       : PropTypes.number,
    temperatureLowLabel  : PropTypes.string,
    humidity             : PropTypes.number,
    humidityLabel        : PropTypes.string,
    precipProbability    : PropTypes.number,
    precipIntensity      : PropTypes.number,
    precipLabel          : PropTypes.string,
    windSpeed            : PropTypes.number,
    windSpeedLabel       : PropTypes.string,
    windSpeedSuffix      : PropTypes.string,
    uvIndex              : PropTypes.number,
    uvIndexLabel         : PropTypes.string,
    unit                 : PropTypes.oneOf([
        Constants.UNIT_SI,
        Constants.UNIT_IMPERIAL,
    ]),
    noShadow             : PropTypes.bool,
};
