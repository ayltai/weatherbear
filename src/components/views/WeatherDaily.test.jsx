import { mount, } from 'enzyme';
import React from 'react';
import renderer from 'react-test-renderer';

import { Constants, } from '../../Constants';
import { WeatherDaily, } from './WeatherDaily';

const date              = 1587628337;
const summary           = 'Partly Cloudy';
const iconId            = 'partly-cloudy-night';
const iconType          = Constants.TYPE_DARK_SKY;
const temperatureHigh   = 29.7;
const temperatureLow    = 22.4;
const humidity          = 0.77;
const precipProbability = 0.48;
const precipIntensity   = 0.2;
const windSpeed         = 7;
const uvIndex           = 4;
const unit              = Constants.UNIT_SI;

const componentHorizontal = (
    <WeatherDaily
        orientation='horizontal'
        date={date}
        summary={summary}
        iconId={iconId}
        iconType={iconType}
        temperatureHigh={temperatureHigh}
        temperatureLow={temperatureLow}
        humidity={humidity}
        precipProbability={precipProbability}
        precipIntensity={precipIntensity}
        windSpeed={windSpeed}
        windSpeedSuffix='km/h'
        uvIndex={uvIndex}
        unit={unit} />
);

const componentVertical = (
    <WeatherDaily
        orientation='vertical'
        date={date}
        summary={summary}
        iconId={iconId}
        iconType={iconType}
        temperatureHigh={temperatureHigh}
        temperatureLow={temperatureLow}
        humidity={humidity}
        precipProbability={precipProbability}
        precipIntensity={precipIntensity}
        windSpeed={windSpeed}
        windSpeedSuffix='km/h'
        uvIndex={uvIndex}
        unit={unit} />
);

describe('<WeatherDaily />', () => {
    it('renders correctly', () => {
        expect(renderer.create(componentHorizontal).toJSON()).toMatchSnapshot();
    });

    it('renders correctly', () => {
        expect(renderer.create(componentVertical).toJSON()).toMatchSnapshot();
    });

    it('mounts without error', () => {
        mount(componentHorizontal);
    });

    it('mounts without error', () => {
        mount(componentVertical);
    });
});
