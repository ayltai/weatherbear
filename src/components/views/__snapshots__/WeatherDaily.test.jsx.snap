// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<WeatherDaily /> renders correctly 1`] = `
<div
  aria-describedby={null}
  className="MuiGrid-root MuiGrid-container MuiGrid-align-items-xs-center MuiGrid-justify-xs-space-evenly"
  onBlur={[Function]}
  onFocus={[Function]}
  onMouseLeave={[Function]}
  onMouseOver={[Function]}
  onTouchEnd={[Function]}
  onTouchStart={[Function]}
  title={null}
>
  <div
    className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-2"
  >
    <p
      aria-describedby={null}
      className="MuiTypography-root makeStyles-textShadowed-2 MuiTypography-body2 MuiTypography-noWrap MuiTypography-alignCenter"
      onBlur={[Function]}
      onFocus={[Function]}
      onMouseLeave={[Function]}
      onMouseOver={[Function]}
      onTouchEnd={[Function]}
      onTouchStart={[Function]}
      title=""
    >
      Mon
    </p>
    <p
      aria-describedby={null}
      className="MuiTypography-root makeStyles-textShadowed-4 MuiTypography-caption MuiTypography-noWrap MuiTypography-alignCenter"
      onBlur={[Function]}
      onFocus={[Function]}
      onMouseLeave={[Function]}
      onMouseOver={[Function]}
      onTouchEnd={[Function]}
      onTouchStart={[Function]}
      title=""
    >
      🌧48%
    </p>
  </div>
  <div
    className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-2"
  >
    <p
      aria-describedby={null}
      className="MuiTypography-root makeStyles-textShadowed-6 MuiTypography-h4 MuiTypography-noWrap MuiTypography-alignCenter"
      onBlur={[Function]}
      onFocus={[Function]}
      onMouseLeave={[Function]}
      onMouseOver={[Function]}
      onTouchEnd={[Function]}
      onTouchStart={[Function]}
      title=""
    >
      <i
        className="wi wi-night-alt-cloudy"
      />
    </p>
  </div>
  <div
    className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6"
  >
    <p
      aria-describedby={null}
      className="MuiTypography-root makeStyles-textShadowed-8 MuiTypography-caption MuiTypography-noWrap MuiTypography-alignLeft"
      onBlur={[Function]}
      onFocus={[Function]}
      onMouseLeave={[Function]}
      onMouseOver={[Function]}
      onTouchEnd={[Function]}
      onTouchStart={[Function]}
      title=""
    >
      Partly Cloudy
    </p>
  </div>
  <div
    className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-2"
  >
    <p
      aria-describedby={null}
      className="MuiTypography-root makeStyles-textShadowed-10 MuiTypography-body2 MuiTypography-noWrap MuiTypography-alignCenter"
      onBlur={[Function]}
      onFocus={[Function]}
      onMouseLeave={[Function]}
      onMouseOver={[Function]}
      onTouchEnd={[Function]}
      onTouchStart={[Function]}
      title=""
    >
      🔥 30°C
    </p>
    <p
      aria-describedby={null}
      className="MuiTypography-root makeStyles-textShadowed-12 MuiTypography-body2 MuiTypography-noWrap MuiTypography-alignCenter"
      onBlur={[Function]}
      onFocus={[Function]}
      onMouseLeave={[Function]}
      onMouseOver={[Function]}
      onTouchEnd={[Function]}
      onTouchStart={[Function]}
      title=""
    >
      ❄ 22°C
    </p>
  </div>
</div>
`;

exports[`<WeatherDaily /> renders correctly 2`] = `
<div
  aria-describedby={null}
  className="MuiGrid-root MuiGrid-container MuiGrid-direction-xs-column MuiGrid-align-items-xs-center MuiGrid-justify-xs-space-evenly"
  onBlur={[Function]}
  onFocus={[Function]}
  onMouseLeave={[Function]}
  onMouseOver={[Function]}
  onTouchEnd={[Function]}
  onTouchStart={[Function]}
  title={null}
>
  <p
    aria-describedby={null}
    className="MuiTypography-root makeStyles-textShadowed-14 MuiTypography-body2 MuiTypography-noWrap MuiTypography-alignCenter"
    onBlur={[Function]}
    onFocus={[Function]}
    onMouseLeave={[Function]}
    onMouseOver={[Function]}
    onTouchEnd={[Function]}
    onTouchStart={[Function]}
    title=""
  >
    Mon
  </p>
  <p
    aria-describedby={null}
    className="MuiTypography-root makeStyles-textShadowed-16 MuiTypography-h4 MuiTypography-noWrap MuiTypography-alignCenter"
    onBlur={[Function]}
    onFocus={[Function]}
    onMouseLeave={[Function]}
    onMouseOver={[Function]}
    onTouchEnd={[Function]}
    onTouchStart={[Function]}
    title=""
  >
    <i
      className="wi wi-night-alt-cloudy"
    />
  </p>
  <p
    aria-describedby={null}
    className="MuiTypography-root makeStyles-textShadowed-18 MuiTypography-body2 MuiTypography-noWrap MuiTypography-alignCenter"
    onBlur={[Function]}
    onFocus={[Function]}
    onMouseLeave={[Function]}
    onMouseOver={[Function]}
    onTouchEnd={[Function]}
    onTouchStart={[Function]}
    title=""
  >
    🔥 30°C
  </p>
  <p
    aria-describedby={null}
    className="MuiTypography-root makeStyles-textShadowed-20 MuiTypography-body2 MuiTypography-noWrap MuiTypography-alignCenter"
    onBlur={[Function]}
    onFocus={[Function]}
    onMouseLeave={[Function]}
    onMouseOver={[Function]}
    onTouchEnd={[Function]}
    onTouchStart={[Function]}
    title=""
  >
    ❄ 22°C
  </p>
  <p
    aria-describedby={null}
    className="MuiTypography-root makeStyles-textShadowed-22 MuiTypography-caption MuiTypography-noWrap MuiTypography-alignCenter"
    onBlur={[Function]}
    onFocus={[Function]}
    onMouseLeave={[Function]}
    onMouseOver={[Function]}
    onTouchEnd={[Function]}
    onTouchStart={[Function]}
    title=""
  >
    🌧48%
  </p>
</div>
`;
